File: 2021/09/19/max_average_subarray.py
"""
https://leetcode.com/problems/maximum-average-subarray-i/
Author: Andrew Jarombek
Date: 9/19/2021
"""

from typing import List


def find_max_average(nums: List[int], k: int) -> float:
    total = 0

    for i in range(k):
        total += nums[i]

    max_avg = total / k

    for i in range(k, len(nums)):
        total -= nums[i - k]
        total += nums[i]

        max_avg = max(max_avg, total / k)

    return max_avg

File: 2022/04/04/maxAverageSubArray.ts
/**
 * https://leetcode.com/problems/maximum-average-subarray-i/
 * @author Andrew Jarombek
 * @since 4/4/2022
 */

function findMaxAverage(nums: number[], k: number): number {
    let sum = 0;

    for (let i = 0; i < k; i++) {
        sum += nums[i];
    }

    let max = sum;

    for (let i = k; i < nums.length; i++) {
        sum += nums[i];
        sum -= nums[i - k];
        max = Math.max(max, sum);
    }

    return max / k;
}